lib/test.js

```diff
@@ -234,6 +244,23 @@ class Test {
 		this.planStack = planStack;
 	}
 
+	timeout(ms) {
+		clearTimeout(this.timeoutTimer);
+		this.timeoutTimer = nowAndTimers.setTimeout(() => {
+			this.saveFirstError(new Error('Test timeout exceeded'));
+
+			if (this.finishDueToTimeout) {
+				this.finishDueToTimeout();
+			}
+		}, ms);
```

Comment:
When the test finishes or fails, the timeout should be cleared to ensure it doesn't keep the worker process alive unnecessarily. Though, at the moment the process is exited when all tests have completed, see #1718 for a proposal to change that.